### DISCUSSION TITLE
added language package but Language table is empty

### DISCUSSION THREAD

**Discussion by artmedia** [oqtane/oqtane.framework#1893] [General] [answered]
I'm working in localhost environment and I've just created a new language package for it-IT
After upgrading Oqtane to v.3.0.1, I added a language package it-IT.
Steps:
1) copied subfolders in Oqtane.Client\Resources as shown in oqtane.translations
![1](https://user-images.githubusercontent.com/4210561/147087218-8ec297e9-5da6-4aba-8294-c47265785b88.png)

2) edited appsettings.json changing "DefaultCulture" to "it-IT" and adding "SupportedCultures": ["en-US","it-IT"]
![2](https://user-images.githubusercontent.com/4210561/147087235-95268ea4-b1fe-4d57-8a14-e901f89f2dca.png)

3) recompiled Oqtane
4) I can see Admin Dashboard in italian lang but there is not Languages Management icon and **Language table** in Oqtane database is **empty** 
![3](https://user-images.githubusercontent.com/4210561/147087486-4bf888e7-2007-4abd-bb88-256f549e97ca.png)

ps. Oqtane.Client.it-IT.nuspec is updated to v.3.0.1
![4](https://user-images.githubusercontent.com/4210561/147088392-4baac068-0b8a-43dc-8415-5bd312fe739e.png)

Am I skipping any steps?
Tnx in advance

**Comment by sbwalker** [comment, marked as answer]
@artmedia some suggestions:

1. It appears that when the Language Management feature was added, the logic was not added to UpgradeManager to automatically add the page/module to each site during upgrade - so on your site it is missing. You can resolve this manually by creating a page which is a child of the Admin page and then adding the Language Management module to that page ( it is in the Admin Module category in the Control Panel ). If you have multiple sites in your installation you will need to do this on each site.

2. I believe the SupportedCultures setting in appsettings.json is no longer used. Instead, during startup the framework automatically determines the supported cultures based on the satellite resource files that are in the /bin. The DefaultCulture setting is still used however rather than setting it in the appsettings.json you can also set it for each site in the Language Management area ( mentioned above ).

There is a dedicated repo for language related topics: https://github.com/oqtane/oqtane.translations

**Reply by artmedia** [reply]
Shaun,
thanks, following your suggestion now it works correctly.
I've just created a pull request for it-IT lang in oqtane.translations.

ps. I noticed that some messages shown to users are not translated (there is no key), eg. "Please Check The Email Address Associated To Your User Account For A Password Reset Notification", "Please Enter The Username Related To Your Account And Then Select The Forgot Password Option Again" in Admin/Login

**Reply by leigh-pointer** [reply]
@artmedia

>  I noticed that some messages shown to users are not translated (there is no key),

 what version of the Oqtane framework are you reporting this from?

**Reply by leigh-pointer** [reply]
Resx fix PR Missing Resx Keys #1897

**Comment by hishamco** [comment]
@sbwalker any action needed regarding your above comments?
